tests/external_command_helper_ut/external_command_helper_ut.c

```diff
@@ -51,6 +74,105 @@ static void on_umock_c_error(UMOCK_C_ERROR_CODE error_code)
     ASSERT_FAIL("umock_c reported error :%" PRI_MU_ENUM "", MU_ENUM_VALUE(UMOCK_C_ERROR_CODE, error_code));
 }
 
+static const char* test_command = "my_program.exe -arg1 -arg2";
+
+#define test_line1 "line1"
+#define test_line2 "another line"
+#define test_line3 "something else"
+#define test_line4 "42"
+#define test_line5 "**&(#(*$&#@(*$"
+
+TEST_DEFINE_ENUM_TYPE(EXTERNAL_COMMAND_RESULT, EXTERNAL_COMMAND_RESULT_VALUES);
+
+static char temp_file_name[L_tmpnam];
+static FILE* last_opened_file_handle;
+static bool pclose_is_pending;
+
+static int pclose_override_return;
+static int hook_pclose(FILE* stream)
+{
+    ASSERT_IS_TRUE(pclose_is_pending, "mocked version of pclose should not have been called without a matching popen");
+
+    // Close and delete the temp file
+    int result = fclose(stream);
+    remove(temp_file_name);
+
+    ASSERT_ARE_EQUAL(int, 0, result);
+
+    LogInfo("Deleted temp file: %s", temp_file_name);
+
+    last_opened_file_handle = NULL;
+    pclose_is_pending = false;
+
+    return pclose_override_return;
+}
+
+static char* test_data_to_report_as_output;
+
+static FILE* hook_popen(
+    char const* command,
+    char const* mode)
+{
+    FILE* result;
+
+    (void)command;
+    (void)mode;
+
+    ASSERT_IS_FALSE(pclose_is_pending, "mocked version of popen only supports one call at a time");
+    pclose_is_pending = true;
+
+    // Pick a file name
+    ASSERT_IS_NOT_NULL(tmpnam(temp_file_name));
+
+    // Open file to fill with test data
+    FILE* temp = fopen(temp_file_name, "w");
+
+    int fputs_result = fputs(test_data_to_report_as_output, temp);
+    ASSERT_IS_TRUE(fputs_result >= 0);
+
+    int fclose_result = fclose(temp);
+    ASSERT_ARE_EQUAL(int, 0, fclose_result);
+
+    // Open file for caller
+    result = fopen(temp_file_name, mode);
+    last_opened_file_handle = result;

```

Comment:
Same here

Reply:
This is saving the handle. And now I see it's not being used like it was in the other tests from block_storage_2 (but it should be used). Fixing

---
In reply to: [554284116](https://github.com/Azure/c-util/pull/93#discussion_r554284116) [](ancestors = 554284116)